### PR TITLE
Limit the thickness of spans in the minimap

### LINKED ISSUE #370: Limit the thickness of spans in the minimap
In the screenshot below, the spans in the minimap are about 4x thicker than the spans in the main timeline. I think it looks very weird aesthetically and might be confusing to new users. Maybe we can set a limit on the max thickness of the spans to be no thicker than in the main timeline view?

image

### PR DESCRIPTION
**Pull request by rubenvp8510**
## Which problem is this PR solving?
- Fixes #370 

## Short description of the changes
- This imposes a limit of the thickness of spans in the minimap. (not more than the span in the main view)

### CODE REVIEW

File: packages/jaeger-ui/src/components/TracePage/TracePageHeader/SpanGraph/render-into-canvas.tsx [hunk 1]
```diff
@@ -36,7 +37,9 @@ export default function renderIntoCanvas(
   canvas.width = cWidth;
   // eslint-disable-next-line no-param-reassign
   canvas.height = cHeight;
-  const itemHeight = Math.max(MIN_ITEM_HEIGHT, cHeight / items.length);
+  const computedItemHeight = Math.max(MIN_ITEM_HEIGHT, cHeight / items.length);
+  const itemHeight = computedItemHeight > MAX_ITEM_HEIGHT ? MAX_ITEM_HEIGHT : computedItemHeight;
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/TracePage/TracePageHeader/SpanGraph/render-into-canvas.tsx]: Why mix ?: and max?

w = Math.min(MAX, Math.max(MIN, expr))

Comment (rubenvp8510) [reply packages/jaeger-ui/src/components/TracePage/TracePageHeader/SpanGraph/render-into-canvas.tsx]: I just wanted to be more explicit, but I think using math.min and math.max is more elegant. I'll change.

Comment (rubenvp8510) [reply packages/jaeger-ui/src/components/TracePage/TracePageHeader/SpanGraph/render-into-canvas.tsx]: Done

File: packages/jaeger-ui/src/components/TracePage/TracePageHeader/SpanGraph/render-into-canvas.test.js [hunk 2]
```diff
@@ -124,16 +125,19 @@ describe('renderIntoCanvas()', () => {
...
         ...items.map((item, i) => {
           const { valueWidth, valueOffset } = item;
           const color = expectedColors[i].output;
           const fillStyle = `rgba(${color.concat(ITEM_ALPHA).join()})`;
-          const height = MIN_TOTAL_HEIGHT / items.length;
+          const height = Math.min(MAX_ITEM_HEIGHT, Math.max(MIN_ITEM_HEIGHT, cHeight / items.length));
           const width = valueWidth / totalValueWidth * getCanvasWidth();
           const x = valueOffset / totalValueWidth * getCanvasWidth();
-          const y = height * i;
+          const y = cHeight / items.length * i;
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/TracePage/TracePageHeader/SpanGraph/render-into-canvas.test.js]: Can we have the same min/max logic in the test? It's easier to read.

Comment (rubenvp8510) [reply packages/jaeger-ui/src/components/TracePage/TracePageHeader/SpanGraph/render-into-canvas.test.js]: Actually the test has the min/max logic,

Comment (rubenvp8510) [reply packages/jaeger-ui/src/components/TracePage/TracePageHeader/SpanGraph/render-into-canvas.test.js]: I'll attach screenshots with small trace. But I think if we limit the height of the minimap for this case we will end with a really small minimap  (not too much aesthetic IMHO) but I'll attach both screenshots.

Comment (tiffon) [reply packages/jaeger-ui/src/components/TracePage/TracePageHeader/SpanGraph/render-into-canvas.test.js]: Thanks

**Comment by rubenvp8510** [comment]
I attached 3 possible implementations here.

1) This PR , limit the thickness of the span bar

Screenshot from 2019-04-23 10-23-19
 

2) Limit the canvas for render (but not the minimap height)

Screenshot from 2019-04-23 11-01-46

3) Also limit the heigh of the minimap according to the number of spans.


Screenshot from 2019-04-23 11-05-07

**Comment by yurishkuro** [comment]
the last two screenshots look wrong - if we reduce the height of the minimap, we should still calculate it as MIN(max_allowed_height, default_span_height * span_count), where default_span_height is the same as in the main timeline view.

**Comment by rubenvp8510** [comment]
Ha! I did the calculations using MIN(max_allowed_height, default_span_height * span_count) but I used the MIN_ITEM_HEIGHT as a default_span_height, that explains why it looks bad.

If I understand correctly default_span_height should be the height of the span in the main timeline. Ok I'll update it.

**Comment by rubenvp8510** [comment]
Screenshot from 2019-04-23 13-01-20
@yurishkuro is this way correct?

Thanks for review!

**Comment by yurishkuro** [comment]
that's what I was suggesting, but now that I see it, it doesn't look particularly great (probably because there's no whitespace between spans).

@tiffon what are your creative suggestions?

**Comment by tiffon** [comment]
1 looks the great, IMO. Is that the one currently set to be merged?

**Comment by rubenvp8510** [comment]
Yes, option 1 is implemented in this PR.

**Pull request merged**